### PR TITLE
Prepare v1.0 release

### PR DESCRIPTION
**Pull request by mathias-arm**
Before this can be submitted  the following PRs need to be merged: 
- https://github.com/veracruz-project/bincode/pull/1
- https://github.com/veracruz-project/bytes/pull/1
- https://github.com/veracruz-project/erased-serde/pull/1
- https://github.com/veracruz-project/indexmap/pull/1
- https://github.com/veracruz-project/json/pull/1
- https://github.com/veracruz-project/log/pull/1
- https://github.com/veracruz-project/nitro-enclave-attestation-document/pull/3
- https://github.com/veracruz-project/num-bigint/pull/2
- https://github.com/veracruz-project/num-integer/pull/1
- https://github.com/veracruz-project/num-rational/pull/1
- https://github.com/veracruz-project/parity-wasm/pull/1
- https://github.com/veracruz-project/ring/pull/3
- https://github.com/veracruz-project/rust/pull/2
- https://github.com/veracruz-project/rust-optee-trustzone-sdk/pull/1
- https://github.com/veracruz-project/rust-protobuf/pull/1
- https://github.com/veracruz-project/rustls/pull/1
- https://github.com/veracruz-project/sct.rs/pull/1
- https://github.com/veracruz-project/typetag/pull/1
- https://github.com/veracruz-project/wasi-types/pull/7
- https://github.com/veracruz-project/wasmi/pull/1
- https://github.com/veracruz-project/webpki/pull/2
- https://github.com/veracruz-project/webpki-roots/pull/1

### CODE REVIEW

**Comment by egrimley-arm** [comment]
Perhaps modify `.github/workflows/main.yml` to read `branches: [ veracruz-v1 ]` instead of `branches: [ main ]`?

Unless GitHub only looks at that file on the default branch ...

**Review by egrimley-arm — APPROVED** [review]
Linux test is failing because of certificate expiry, perhaps fixed by PR #285.

**Review by dominic-mulligan-arm — DISMISSED** [review]
...that was a lot of `Cargo.lock` files 🥲

**Comment by mathias-arm** [comment]
The need for platform-specific `Cargo.lock` files does not seem to be needed anymore.

**Review by dominic-mulligan-arm — APPROVED** [review]

**Review by dominic-mulligan-arm — APPROVED** [review]

**Comment by dominic-mulligan-arm** [comment]
+1 reached, merging.

**Comment by dominic-mulligan-arm** [comment]
Oof, spoke too soon: @mathias-arm not all of the commits are signed, so merging is blocked.  Can you rewrite history to sign them and I'll re-approve immediately?

**Comment by mathias-arm** [comment]
Sorry about that. I just pushed an update with all commits signed.

**Review by dominic-mulligan-arm — APPROVED** [review]

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -190,12 +211,20 @@ jobs:
       - name: Build and test Veracruz on IceCap
         run: bash /work/veracruz/icecap/ci/run.sh
 
+     # Upload Cargo.lock files, using the action from github
+      - name: Upload Cargo.lock files
```
Comment (geky) [inline .github/workflows/main.yml:215]: Just a note that "workflow artifacts" and "release artifacts" are handled a bit differently.

> By default, GitHub stores build logs and [workflow] artifacts for 90 days

If we want these artifacts to persist they will need to eventually be attached to a release

**Review by geky — APPROVED** [review]
Looks good to me

**Comment by dominic-mulligan-arm** [comment]
+1+1 = +1 reached, merging.

File: .github/workflows/main.yml (hunk 1 above)
Comment (geky) [reply .github/workflows/main.yml:215]: Ah got it, a good use of artifacts then

**Pull request merged**